YLDA_HOME environment variable

deanmalmgren opened the pull request:
Hey shravanmn,

I love the YLDA code! I've been using it for a project and I have been impressed by how much better it is than the Mahout LDA implementation both in terms of speed and setup time. Thanks for making this public! I just pushed a couple of minor changes that I needed to do to get things working for my purposes and I thought I would let you know in case you want to pull them into the master branch:
1. I added a YLDA_HOME environment variable to the runLDA.sh script make it possible to run YLDA from outside of the source root (something that I need to do). 
2. I was having wierd syntax problems on my system with the LDA.sh and runLDA.sh scripts, so I changed these to being /bin/bash based rather than /bin/sh based.

Cheers.
Dean

Commit:
changed LDA.sh and runLDA.sh to using /bin/bash rather than /bin/sh, which was giving me problems. added .gitignore file to repository root.

Commit:
for real this time, i added .gitignore file.

Commit:
made it possible to specify YLDA_HOME environment variable so that it is not necessary to run YLDA code from the YLDA root.

shravanmn commented:
Please see in line..

On Tuesday 06 September 2011 09:28 PM, deanmalmgren wrote:

> Hey shravanmn,
> 
> I love the YLDA code! I've been using it for a project and I have been impressed by how much better it is than the Mahout LDA implementation both in terms of speed and setup time. Thanks for making this public! I just pushed a couple of minor changes that I needed to do to get things working for my purposes and I thought I would let you know in case you want to pull them into the master branch:
> 
> Thanks for the good word!

Sure we will be glad to pull in any changes that make things simpler.

> 1. I added a YLDA_HOME environment variable to the runLDA.sh script make it possible to run YLDA from outside of the source root (something that I need to do).
> 
> Can you expand a bit more on what you mean by outside of the source root?
> 2. I was having wierd syntax problems on my system with the LDA.sh and runLDA.sh scripts, so I changed these to being /bin/bash based rather than /bin/sh based.
> 
> Oh can you tell me what the problems were and what is your system 
> configuration? We have only tested system dependencies so much. Will be 
> good to see how they can be fixed.

Thanks,
--Shravan

> Cheers.
> Dean

deanmalmgren commented:
also in line...but before I get to that I thought I should mention that 
if there is a better way to implement these things, please let me know.

Thanks again!
Dean

On 2011/09/07 22:52, shravanmn wrote:

> 1. I added a YLDA_HOME environment variable to the runLDA.sh script make it possible to run YLDA from outside of the source root (something that I need to do).
> 
> Can you expand a bit more on what you mean by outside of the source root?

For example, if we have the YLDA code installed at /usr/lib/ylda and we 
are running a script which programatically invokes YLDA from 
/some/other/directory, the problem is that the runLDA.sh script assumes 
that you are in /usr/lib/ylda (it sources things, etc.). By adding the 
YLDA_HOME environment variable it is easy to call runLDA.sh elsewhere.

> > 1. I was having wierd syntax problems on my system with the LDA.sh and runLDA.sh scripts, so I changed these to being /bin/bash based rather than /bin/sh based.
> > 
> > Oh can you tell me what the problems were and what is your system
> > configuration? We have only tested system dependencies so much. Will be
> > good to see how they can be fixed.

I have tried this on fresh installs of Ubuntu 10.04 and 11.04, both 
running on a laptop and on an EC2 cluster. Apparently dash is Ubuntu's 
replacement for sh and dash is not too fond of "==" for string comparison.

shravanmn commented:
Thanks Dean. This will be useful. Before I can pull this, I need a contribution license agreement signed. You can find the agreement here https://github.com/shravanmn/Yahoo_LDA/raw/master/Contribution_License_Agreement_Yahoo.pdf

If you have trouble signing this document in pdf let me know we can figure out alternate ways to that.

deanmalmgren commented:
Hi Shravan,

Sorry for the belated reply. I had a crunch-time deadline the past few 
weeks and I did not have a chance to adequately review the agreement 
before today. Attached is the signed contribution license agreement, 
giving myself and Mike Stringer the authority to contribute to the YLDA 
source on behalf of Datascope Analytics.

If there is anything else you need from us, please let me know.

Cheers.
Dean

On 2011/09/13 03:56, shravanmn wrote:

> Thanks Dean. This will be useful. Before I can pull this, I need a contribution license agreement signed. You can find the agreement here https://github.com/shravanmn/Yahoo_LDA/raw/master/Contribution_License_Agreement_Yahoo.pdf
> 
> If you have trouble signing this document in pdf let me know we can figure out alternate ways to that.

## 

_dean malmgren_
+1.312.884.9214

Commit:
added soft link to LDALibs -> . so things will actually run.
